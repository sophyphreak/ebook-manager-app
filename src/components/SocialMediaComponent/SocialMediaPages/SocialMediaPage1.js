import React, { Component } from 'react';
import { Card, CardHeader, CardBody, Button, Form, Label } from 'reactstrap';

import AmazonUrl from '../../FormElements/AmazonUrl';
import NonFictionOrFiction from '../../FormElements/NonFictionOrFiction';
import Genre from '../../FormElements/Genre';
import SubGenre from '../../FormElements/SubGenre';
import Email from '../../FormElements/Email';

const SocialMediaPage1 = ({
  error,
  onSubmit,

  amazonUrl,
  onAmazonUrlChange,

  nonFictionOrFiction,
  onNonFictionOrFictionChange,

  genre,
  onGenreChange,

  subGenre,
  onSubGenreChange,

  email,
  onEmailChange
}) => (
  <Card className="animated fadeIn">
    <CardHeader>
      <h4>
        <strong>Social Media Submission - 1 of 3</strong>
      </h4>
    </CardHeader>
    <CardBody>
      <Form onSubmit={onSubmit}>
        <AmazonUrl
          label="Amazon URL*"
          value={amazonUrl}
          onChange={onAmazonUrlChange}
          hasError={!!error.message}
          errorMessage={error.amazonUrl}
        />
        <NonFictionOrFiction
          nonFictionOrFiction={nonFictionOrFiction}
          onNonFictionOrFictionChange={onNonFictionOrFictionChange}
        />
        {nonFictionOrFiction === 'Fiction' && (
          <div className="animated fadeIn">
            <Genre
              label="Genre:*"
              value={genre}
              onChange={onGenreChange}
              hasError={!!error.message}
              errorMessage={error.genre}
            />
          </div>
        )}
        <SubGenre
          label="SubGenre:"
          subGenre={subGenre}
          onSubGenreChange={onSubGenreChange}
          hasError={!!error.message}
          errorMessage={error.subGenre}
        />
        <Email
          label="Email:*"
          email={email}
          onEmailChange={onEmailChange}
          hasError={!!error.message}
          errorMessage={error.email}
        />
        {error.message && <p>{error.message}</p>}
        <Button type="submit" size="sm" color="primary" value="Submit">
          Next Section
        </Button>
      </Form>
    </CardBody>
  </Card>
);

export default SocialMediaPage1;
