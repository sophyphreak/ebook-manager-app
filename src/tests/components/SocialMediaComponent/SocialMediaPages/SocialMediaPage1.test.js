import React from 'react';
import { shallow } from 'enzyme';
import SocialMediaPage1 from '../../../../components/SocialMediaComponent/SocialMediaPages/SocialMediaPage1';

let error, wrapper;

beforeEach(() => {
  error = {
    message: ""
  };
  wrapper = shallow(<SocialMediaPage1 error={error} />);
});

test('should render SocialMediaPage1 correctly', () => {
  expect(wrapper).toMatchSnapshot();
});