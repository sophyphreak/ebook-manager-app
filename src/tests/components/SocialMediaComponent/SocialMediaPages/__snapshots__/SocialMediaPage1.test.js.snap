// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SocialMediaPage1 correctly 1`] = `
<Card
  className="animated fadeIn"
  tag="div"
>
  <CardHeader
    tag="div"
  >
    <h4>
      <strong>
        Social Media Submission - 1 of 3
      </strong>
    </h4>
  </CardHeader>
  <CardBody
    tag="div"
  >
    <Form
      tag="form"
    >
      <AmazonUrl
        hasError={false}
        label="Amazon URL*"
      />
      <NonFictionOrFiction
        hasError={false}
      />
      <SubGenre
        hasError={false}
        label="SubGenre:"
      />
      <Email
        hasError={false}
        label="Email:*"
      />
      <Button
        color="primary"
        size="sm"
        tag="button"
        type="submit"
        value="Submit"
      >
        Next Section
      </Button>
    </Form>
  </CardBody>
</Card>
`;
